app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -24,37 +24,55 @@ class PromotedStoryListView @JvmOverloads constructor(
 
   @Inject
   lateinit var bindingInterface: ViewBindingShim
+  private var dataList: List<PromotedStoryViewModel> = listOf()
```

Comment:
Do we need this? I don't think the equality check is needed below since we already do that later in the pipeline when deciding whether to deploy the LiveData (and we didn't use it before with the binding adapter that was being used).

Reply:
Are we able to get the data list from the adapter? We wouldn't want to continue to rebind the view & data if the data hasn't changed, so is there a different way to access the data list?

Reply:
Done & removed, we want to check for adapter creation, not the data list